src/main/java/org/cactoos/iterable/IterableOf.java

```diff
@@ -154,12 +159,20 @@ public boolean equals(final Object other) {
                     () -> other != null,
                     () -> Iterable.class.isAssignableFrom(other.getClass()),
                     () -> {
-                        final Iterable<?> compared = (Iterable<?>) other;
-                        final Iterator<?> iterator = compared.iterator();
-                        return new Unchecked<>(
+                        final Iterable<X> compared = (Iterable<X>) other;
+                        return new ScalarWithFallback<>(
                             new And(
-                                (X input) -> input.equals(iterator.next()),
-                                this
+                                (X value) -> true,
+                                new Matched<>(
```

Comment:
@lxpdd sorry, I just realized there is something strange here, I don't see how that would implement equals, can you add 2 tests of equals for iterables of equal length : one when the two iterables are equals, and one when they are not?

Reply:
@victornoel 
There is two tests that cover it:

**notEqualsToIterableWithDifferentElements** comparing iterables (1, 2) and (1,0) and
**isEqualToIterableWithTheSameElements** comparing iterables (1,2) and (1,2)

Reply:
@lxpdd ok, thanks, I re-read the code from `Matched` and now I understand the logical mistake I made, all is good :) thx!